File: backend/app/models/photo.py
import uuid

from sqlalchemy import DateTime, ForeignKey, String, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class TripPhoto(Base):
    __tablename__ = "trip_photos"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    trip_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("trips.id", ondelete="CASCADE"), nullable=False)
    object_key: Mapped[str] = mapped_column(String, nullable=False)
    file_name: Mapped[str] = mapped_column(String, nullable=False)
    caption: Mapped[str] = mapped_column(String, nullable=False, default="")
    created_at: Mapped[DateTime] = mapped_column(DateTime(timezone=True), server_default=func.now())

File: frontend/types/photo.ts
export interface TripPhoto {
  id: string;
  trip_id: string;
  file_name: string;
  caption: string;
  url: string;
  created_at: string;
}

export interface BookingFile {
  id: string;
  booking_id: string;
  file_name: string;
  file_type: string;
  file_size: number;
  url: string;
  created_at: string;
}
